File: 107 Binary Tree Level Order Traversal II/BinaryTreeLevelOrderTraversal_DFS.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class BinaryTreeLevelOrderTraversal_DFS {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        List<List<Integer>> res = new LinkedList<List<Integer>>();
        dfs(res, root, 0);
        return res;
    }
    
    private void dfs(List<List<Integer>> res, TreeNode root, int level) {
        if (root == null) {
            return;
        }
        if (level >= res.size()) {
            res.add(0, new LinkedList<Integer>());
        }
        dfs(res, root.left, level + 1);
        dfs(res, root.right, level + 1);
        res.get(res.size() - level - 1).add(root.val);
    }
}

File: 107 Binary Tree Level Order Traversal II/binary_tree_level_order_traversal_dfs.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
        res = []
        self.bfs(res, root, 0)
        return res
    
    def bfs(self, res, root, level):
        if (root == None):
            return
        if level >= len(res):
            res.insert(0, [])
        self.bfs(res, root.left, level + 1)
        self.bfs(res, root.right, level + 1)
        res[len(res) - level - 1].append(root.val)